proposals/xxxx-work-graphs.md

```diff
@@ -0,0 +1,455 @@
+
+Full documentation of the Work Graphs feature including DirectX runtime, HLSL, and DXIL documentation is available in the [Work Graphs Spec on GitHub](https://microsoft.github.io/DirectX-Specs/d3d/WorkGraphs.html). This document subsumes that document specifically as it relates to HLSL and the compiler features.
+
+## Motivation
+
+Current language and runtime limitations make generating work on GPU threads insufficient to meet the needs of some workloads. If existing GPU features (like ExecuteIndirect) can't sufficiently generate work the application must generate the work from the CPU resulting in unnecessary round-tripping between the GPU and CPU. Work Graphs solves this problem by enabling more robust GPU-based work creation APIs.
+
+## Proposed solution
+
+Work graphs allows an application to specify a set of tasks as _nodes_ in a graph representing a more complex workload. Each _node_ has a fixed shader which takes one or more _input records_ as input and can produce one or more _output records_ as output.
+
+### Launch Modes
+
+_Node_ shaders have one of three _launch modes_:
+
+* Thread
+* Broadcasting
+* Coalescing
+
+_Thread launch_ nodes represent an individual thread of work that processes a single _input record_. Thread launch nodes use a thread group size of `(1,1,1)`.
+
+_Broadcasting launch_ nodes represent a grid of work operating on a single _input record_. Each input record to a broadcasting node launches a full dispatch grid. The size of the dispatch grid can either be fixed for the node or specified in the input.
+
+_Coalescing launch_ nodes represent a thread group operating on a shared array of _input records_. The shader declares the maximum number of records per thread group.
+
+### Records
+
+_Input records_ represent inputs to node shaders, and _output records_ represent outputs from node shaders. Records can be singular or arrays. _Input records_ can be read-only or read-write, while _output records_ are read-write but uninitialized when created.
+
+## Detailed design
+
+### Node Entry Functions
+
+Node shaders are compute shaders built into library targets. Shader entries annotated as `[shader("node")]` are usable as work graph nodes.
+
+A broadcasting launch mode node entry may also be a compute entry (annotated with `[shader("compute")]`). In this case the node's input is an empty input record.
+
+#### Function Attributes
+
+All node shaders except _thread launch_ nodes must specify the thread group size using the `[numthreads(<x>, <y>, <z>)]` attribute.
+
+Node shaders support all compute shader function annotations, but also have new annotations unique to node shaders.
+
+##### **`[NodeLaunch("<mode>")]`**
+Valid values for node launch mode are `broadcasting`, `coalescing`, or `thread`. If the `NodeLaunch` attribute is not specified on a node entry the default launch mode is `broadcasting`.

```

Comment:
Note: in historically the attribute name (i.e. `NodeLaunch`) is case-insensitive, but the arguments are case-sensitive. We should follow that with Work Graph attributes.